chain/signedtxn.go

```diff
@@ -0,0 +1,343 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/fees"
+	"github.com/ava-labs/hypersdk/internal/emap"
+	"github.com/ava-labs/hypersdk/internal/math"
+	"github.com/ava-labs/hypersdk/internal/mempool"
+	"github.com/ava-labs/hypersdk/keys"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/state/tstate"
+	"github.com/ava-labs/hypersdk/utils"
+
+	internalfees "github.com/ava-labs/hypersdk/internal/fees"
+)
+
+var (
+	_ emap.Item    = (*SignedTransaction)(nil)
+	_ mempool.Item = (*SignedTransaction)(nil)
+)
+
+type SignedTransaction struct {
```

Comment:
I'm not sure how I feel about this naming. I personally get pretty confused seeing `SignedTransaction` and `Transaction` as I keep assuming `Transaction` is signed. We could consider something else like `SignedTransaction` and `UnsignedTransaction` or `Transaction` and `TransactionData`.

Reply:
It looks like we have only several places the use the unsigned transaction. Therefore, I'm not really opinionated here. I've changed the following:
SignedTransaction -> Transaction
Transaction -> TransactionData